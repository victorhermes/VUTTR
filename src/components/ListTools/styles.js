import styled from "styled-components";

export const Container = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
    padding-bottom: 10px;
    margin: 10px;
    width: 100%;
    max-width: 550px;
`;

export const ToolSection = styled.div`
    width: 100%;
    height: auto;
    background-color: #ffffff;
    margin: 10px;
    padding: 15px;
    color: #000;
    border-radius: 5px;

    h1 {
        padding-bottom: 10px;
    }

    div {
        display: flex;
        flex-wrap: wrap;
        padding-top: 15px;

        span {
            list-style: none;
            color: #31225f;
            font-weight: bold;
            padding-right: 10px;
        }
    }
`;

export const ToolHeader = styled.div`
    display: flex;
    justify-content: space-between;
    align-items: flex-start;

    img {
        width: 15px;
    }
`;

export const Header = styled.div`
    display: flex;
    justify-content: space-between;
    width: 100%;
    box-sizing: border-box;
    margin-bottom: 10px;
`;
