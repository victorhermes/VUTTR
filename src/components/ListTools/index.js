import React, { Component } from "react";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import ToolsActions from "../../store/ducks/tools";
import Button from "../Button";
import Search from "../Search";
import IconClose from "./img/Icon-Close.svg";
import { Container, ToolSection, ToolHeader, Header, Tags } from "./styles";

class ListTools extends Component {
    componentDidMount() {
        const { getToolRequest } = this.props;
        getToolRequest();
    }

    render() {
        const { tools } = this.props;
        return (
            <Container>
                <Header>
                    <input
                        type="text"
                        name="text"
                        placeholder="Procurar ferramenta"
                    />
                    <Button />
                </Header>

                {tools.data.length ? (
                    tools.data.map(tool => (
                        <ToolSection key={tool.id}>
                            <ToolHeader>
                                <a href={tool.link} target="_blank">
                                    <h1> {tool.title}</h1>
                                </a>
                                <img src={IconClose} alt="Remover" />
                            </ToolHeader>

                            <p>{tool.description}</p>

                            <Tags>
                                {tool.tags.map(tag => (
                                    <span key={tag}>#{tag}</span>
                                ))}
                            </Tags>
                        </ToolSection>
                    ))
                ) : (
                    <h2 align="center">"Não há nada aqui!</h2>
                )}
            </Container>
        );
    }
}

const mapStateToProps = state => ({
    tools: state.tools
});

const mapDispatchToProps = dispatch =>
    bindActionCreators(ToolsActions, dispatch);

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(ListTools);
