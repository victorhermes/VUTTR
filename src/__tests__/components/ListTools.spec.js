import { mount, shallow } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';
import createStore from 'redux-mock-store';
import styled from 'styled-components';

import 'jest-styled-components';
import ListTools from '~/components/ListTools';

const mockStore = createStore();

const Btn = styled.button`
  font-size: 18px;
  color: #31225fa1;
  background: transparent;
  border: 0;
  border-radius: 5px;
  padding: 3px 5px 3px 5px;
  text-transform: uppercase;
`;

const INITIAL_STATE = {
  tools: {
    data: [
      {
        title: 'eslint-plugin-react',
        link: 'https://github.com/yannickcr/eslint-plugin-react',
        description: 'React specific linting rules for ESLint ',
        id: 1,
        tags: [
          {
            label: 'modal',
            value: 'modal',
          },
          {
            label: 'rodal',
            value: 'rodal',
          },
          {
            label: 'react',
            value: 'react',
          },
        ],
      },
      {
        title: 'yup',
        link: 'https://github.com/yup',
        description: 'Yup is something cool!',
        id: 2,
        tags: [
          {
            label: 'modal',
            value: 'modal',
          },
          {
            label: 'rodal',
            value: 'rodal',
          },
          {
            label: 'react',
            value: 'react',
          },
        ],
      },
    ],
  },
};

const store = mockStore(INITIAL_STATE);

it('Should render the tools', () => {
  const wrapper = mount(
    <Provider store={store}>
      <ListTools />
    </Provider>,
  );

  const button = shallow(<Btn />);
  expect(wrapper.find('h1').length).toBe(2);
  expect(wrapper.find('p').length).toBe(3);
  expect(button).toMatchSnapshot();
  expect(wrapper.find('button').length).toBe(5);
  expect(wrapper.find('span').length).toBe(6);
});
