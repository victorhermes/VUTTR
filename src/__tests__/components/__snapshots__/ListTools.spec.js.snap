// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the tools 1`] = `ShallowWrapper {}`;
